### ISSUE TITLE
Second AsyncWebServer on port 81

### ISSUE THREAD

**Issue by JOHugo** [oferzv/wifiTool#3]
My second web server doesn't work if I run wifiTool.runWifiPortal()

#include <wifiTool.h>
WifiTool wifiTool;
AsyncWebServer server2(81);

void setup() {
  Serial.begin(115200);
  Serial.println("System started");
  wifiTool.begin();
  wifiTool.runWifiPortal();

  server2.on("/", HTTP_GET, [] (AsyncWebServerRequest *request) {
    request->send(200, "text/plain", "Web server2");
  }); 
  server2.begin();
}

void loop() {
  // put your main code here, to run repeatedly:
}

**Comment by gonesurfing** [comment]
I'm having a similar issue resetting the server and starting wifiportal. Any luck with this?

**Comment by talofer99** [comment]
The wifiTool.runWifiPortal(); will over ride any other settings,
It was attended to be used when pressing a button for example.

Note that you can check if you got connected on the setup 

if (!wifiTool.wifiAutoConnect())
  {
    Serial.println("fail to connect to wifi!!!!");
    wifiTool.runApPortal();
  }

**Comment by gonesurfing** [comment]
Here is a code snippet to reproduce it. I am using it as a "button", but none of the server handlers are reset, or wifiportal handlers added.

`#include <AsyncTCP.h>
#include <ESPAsyncWebServer.h>
#include <SPIFFS.h>
#include <wifiTool.h>
WifiTool wifiTool;

AsyncWebServer server(80);

volatile bool requestPortal = false;
void setup(void) {
  Serial.begin(115200);
  Serial.println("System started");

  // Initialize SPIFFS
  if(!SPIFFS.begin()){
    Serial.println("An Error has occurred while mounting SPIFFS");
    return;
  }

  wifiTool.begin(false);
  if (!wifiTool.wifiAutoConnect()) {
    Serial.println("fail to connect to wifi!!!!");
    wifiTool.runApPortal();
  }

  // Route for root / web page
  server.on("/", HTTP_GET, [](AsyncWebServerRequest *request){
    request->send(SPIFFS, "/index.html");
  });
  server.on("/portal", HTTP_GET, [](AsyncWebServerRequest *request){
    request->send_P(200, "text/plain", "OK");
    Serial.println("WifiTool Portal Request");
    requestPortal = true;
  });

  // Start server
  server.begin();
}

void loop(void) {
  //do some work here
  
  if (requestPortal) {
    Serial.println("Starting wifi portal in 2s");
    delay(2000);
    wifiTool.runWifiPortal();
  }
}`

**Comment by gonesurfing** [comment]
I finally got it to work. I needed "server.reset()" and "server.end()". Also without the 1s delay, I would get a crash/reboot.

if (requestPortal) {
    Serial.println("Starting wifi portal in 1s");
    server.reset();
    server.end(); 
    delay(1000); //otherwise things crash
    wifiTool.runWifiPortal();
  }

**Comment by JOHugo** [comment]
> I'm having a similar issue resetting the server and starting wifiportal. Any luck with this?

I've changed my stuff to only use one AsyncWebServer. I've change the wifitool library to call index.html (my web page on SPIFFS) by default, instead of wifi_index.html. From index.html I have a button to call wifi_index.html. This is all I need on the ESP32 and all my HTML files are on SPIFFS. I really like this.

#include <wifiTool.h>
WifiTool wifiTool;

void setup() {
  Serial.begin(115200);
  Serial.println("System started");

  wifiTool.begin();
  wifiTool.runWifiPortal();
}

void loop() {
  // put your main code here, to run repeatedly:

}

**Comment by gonesurfing** [comment]
> I've changed my stuff to only use one AsyncWebServer. I've change the wifitool library to call index.html (my web page on SPIFFS) by default, instead of wifi_index.html. From index.html I have a button to call wifi_index.html. This is all I need on the ESP32 and all my HTML files are on SPIFFS. I really like this.

Good idea, except for my case I do sensor processing in the loop(). Wince wifiTool.runWifiPortal() blocks, that won't work for my case.

**Comment by JOHugo** [comment]
Oops, adding sensors was next on my list. Will take a look at your solution.
